frontend/src/middleware/search.ts

```diff
@@ -0,0 +1,25 @@
+import { useSearchStore } from "~/stores/search"
+
+import type { Middleware } from "@nuxt/types"
+
+export const searchMiddleware: Middleware = ({ redirect, route, $pinia }) => {
+  const {
+    query: { q, searchBy },
+  } = route
+  /**
+   * This anonymous middleware redirects any search without a query to the homepage.
+   * This is meant to block direct access to /search and all sub-routes, with
+   * an exception for the 'creator' filter. The creator filter doesn't send
+   * the search query to the API.
```

Comment:
> The creator filter doesn't send the search query to the API.

Is this true? How does it work, then? :open_mouth:

Reply:
Thank you for noticing this. I realized that this was implemented incorrectly: even if we search by creator (which we currently can only do by manually updating the URL, the UI for searchBy creator was removed), the frontend URL query parameter is `q`: `?q=cat&searchBy=creator` is converted to `?searchBy=creator&creator=cat` API request query. I've removed the check for `searchBy` parameter, and updated the comment accordingly.